### DISCUSSION TITLE
How to pass an object in string without convert it to typescript types?

### DISCUSSION THREAD

**Discussion by CursedMun** [ivanhofer/typesafe-i18n#477] [Q&A]
like {foo: '{test: {'1': '1'}}'}

**Comment by ivanhofer** [comment]
Hi @CursedMun, can you please share more details on what you are trying to achieve?

**Reply by CursedMun** [reply]
Hi, so basically I need to be able to store JSON objects in the format of a string without converting all the brackets to arguments

**Reply by ivanhofer** [reply]
This is currently not possible with `typesafe-i18n`. I always wondered how long it would take until someone asks for this ^^
As `json` is a format for development, your translations could be probably modeled in a way where you don't have json structure in the translation. A detailed example would help to understand your needs better.

If you really need it, you could use following workaround:

_src/i18n/json.ts_
```ts
export let renderJsonInTranslations = false

export const jsonPart = (jsonString: string) => renderJsonInTranslations ? jsonString : '[JSON]'
```

_src/i18n/en/index.ts_
```ts
import type { BaseTranslation } from '../i18n-types'
import { jsonPart } from '../json'

const en: BaseTranslation = {
   foo: `Hello ${jsonPart('{ test: { "1": "1" } }')}`
}

export default en
```

You would need to wrapp all those `json` strings into the `jsonPart` helper function.

When `typesafe-i18n` loads the translations, `renderJsonInTranslations` will be false and the text `'[JSON]'`will be emitted (This has no effect on the typesafety. It is just here for you to indicate that this is a placeholder value).

Then in your application code, at the earliest possible point (before importing anything from `typesafe-i18n`) set `renderJsonInTranslations` to `true`. That way `jsonPart` will output the original text and you will see the `json` structure in your translations.

It is not ideal, but it is possible to achieve with the current version.

**Reply by CursedMun** [reply]
Yeah, that fits my needs, I can obviously make that without the need of the json, but it becomes very much a spaghetti of code for a big project where all the answers are returned via json.

**Reply by ivanhofer** [reply]
This sounds like you are mixing tranlations and content. But without any details I can't help you find a better approach

**Comment by CursedMun** [comment]


**Reply by ivanhofer** [reply]
what do you mean?
